src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -525,6 +547,15 @@ public int getMaxTotalUses() {
         return maxTotalUses;
     }
 
+    public List<NodePropertyDescriptor> getNodePropertyDescriptors() {
+        return NodePropertyDescriptor.for_(NodeProperty.all(), EC2AbstractSlave.class);
+    }
+
+    public DescribableList<NodeProperty<?>, NodePropertyDescriptor> getNodeProperties() {
+        assert nodeProperties != null;
+    	return nodeProperties;
```

Comment:
```suggestion
    	return Objects.requireNonNull(nodeProperties);
```

Then you can remove the assertion above.